ad-you-like/ad-you-like-htb.js

```diff
@@ -0,0 +1,576 @@
+         * response calls it once it has completed.
+         *
+         * If your endpoint does not support passing in an ID, simply use
+         * Partner.CallbackTypes.CALLBACK_NAME and the wrapper will take care of handling request
+         * matching by generating unique callbacks for each request using the callbackId.
+         *
+         * If your endpoint is ajax only, please set the appropriate values in your profile for this,
+         * i.e. Partner.CallbackTypes.NONE and Partner.Requesttypes.AJAX. You also do not need to provide
+         * a callbackId in this case because there is no callback.
+         *
+         * The return object should look something like this:
+         * {
+         *     url: 'http://bidserver.com/api/bids' // base request url for a GET/POST request
+         *     data: { // query string object that will be attached to the base url
+         *        slots: [
+         *             {
+         *                 placementId: 54321,
+         *                 sizes: [[300, 250]]
+         *             },{
+         *                 placementId: 12345,
+         *                 sizes: [[300, 600]]
+         *             },{
+         *                 placementId: 654321,
+         *                 sizes: [[728, 90]]
+         *             }
+         *         ],
+         *         site: 'http://google.com'
+         *     },
+         *     callbackId: '_23sd2ij4i1' //unique id used for pairing requests and responses
+         * }
+         */
+
+        /* ---------------------- PUT CODE HERE ------------------------------------ */
+        var queryObj = {};
+        var callbackId;
+        var partnerId = null;
+        var liveRampIp = null;
+        var adserverOrgIp = null;
+
+        var bids = returnParcels.reduce(function (accumulator, bid) {
+            var bidId = bid.htSlot.getId();
+
+            if (!callbackId) {
+                callbackId = bidId;
+            }
+
+            var sizesArray = bid.xSlotRef.sizes;
+            var size = sizesArray[0];
+            accumulator[bidId] = {};
+            accumulator[bidId].PlacementID = bid.xSlotRef.placementId;
+            accumulator[bidId].Width = size.width;
+            accumulator[bidId].Height = size.height;
+            accumulator[bidId].AvailableSizes = formatAvailableSize(sizesArray);
+
+            if (!partnerId) {
+                partnerId = bid.xSlotRef.partnerId;
+            }
+
+            if (!liveRampIp && bid.identityData) {
+                liveRampIp = bid.identityData.LiveRampIp;
+            }
+
+            if (!adserverOrgIp && bid.identityData) {
+                adserverOrgIp = bid.identityData.AdserverOrgIp;
+            }
+
+            return accumulator;
+        }, {});
+
+        /* Change this to your bidder endpoint. */
+        var baseUrl = Browser.getProtocol() + '//hb-api.omnitagjs.com/hb-api/ix/v1?';
+        baseUrl += 'RefererUrl=' + encodeURIComponent(Browser.getReferrer());
```

Comment:
Include query params as part of the query object , 
You can leave the baseUrl without the query params 
i.e baseUrl should just be  `Browser.getProtocol() + '//hb-api.omnitagjs.com/hb-api/ix/v1`;
Any query params you want to add should be part of the query object. 

This is causing your system tests to fail, so you can verify your fix by running the system test ... 

https://kb.indexexchange.com/ix_library_partners/build_adapter/run_the_adapter_test_cases.htm

Reply:
Will do.  Thanks

Reply:
I've made the change but every try to run the debug script will result in an error from the adapter-development-suite.
Seems to happen on other adapters too.


`> ht-wrapper-adapters@1.0.0 debug
> node ./node_modules/adapter-development-suite/bin/index.js

/home/user/path/ix-library-adapters/node_modules/adapter-development-suite/bin/index.js:86
    if (cwd.length - cwd.lastIndexOf('ht-wrapper-adapters') === repoNameLength
            ^

TypeError: Cannot read property 'length' of undefined
    at Object.<anonymous> (/home/user/path/ix-library-adapters/node_modules/adapter-development-suite/bin/index.js:86:13)
    at Module._compile (node:internal/modules/cjs/loader:1083:30)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1112:10)
    at Module.load (node:internal/modules/cjs/loader:948:32)
    at Function.Module._load (node:internal/modules/cjs/loader:789:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:72:12)
    at node:internal/main/run_main_module:17:47
npm ERR! code 1
npm ERR! path /home/user/path/ix-library-adapters
npm ERR! command failed
npm ERR! command sh -c node ./node_modules/adapter-development-suite/bin/index.js

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/guillaume/.npm/_logs/2021-02-04T16_18_56_047Z-debug.log
`

Reply:
Hi , which repo and folder are you running this from ?

```
ht-wrapper-adapters@1.0.0 debug
node ./node_modules/adapter-development-suite/bin/index.js
/home/user/path/ix-library-adapters/node_modules/adapter-development-suite/bin/index.js:86
if (cwd.length - cwd.lastIndexOf('ht-wrapper-adapters') === repoNameLength
^
TypeError: Cannot read property 'length' of undefined
```

Reply:
Hello IX team. 
We have been advised we would be included on your "next sprint". Can you please let us know when that will be ?
I'm still waiting for an answer to make sure everything is fine with our adapter?

Reply:
Hi Guiann 
Yes, this work has been included in this sprint and we are actively working on adding your adapter.

Reply:
Thank you,

when running  'npm run debug'  from our adapter folder I don't see any error anymore. And I'm able to display an ad in the localhost test page.

Reply:
Yes confirming, no errors there .

Reply:
Thank you